index.js

```diff
@@ -169,7 +169,7 @@ jasmine.Expectation.prototype.wrapCompare = function(name, matcherFactory) {
 
     if (!webdriver.promise.isPromise(expectation.actual) &&
         !webdriver.promise.isPromise(expected)) {
```

Comment:
Why even have the no promises case if we're going to pass everything to `compare` anyway?